### PR TITLE
Fix network documentation coverage

### PR DESCRIPTION
**Pull request by jukkar**
This should improve network documentation coverage percentage.
So I added missing doxygen comments when applicable or hide internal documentation which does not need to be documented.
There are lot of commits, but most of them are very small. Each header file changed is placed in its own commit for easier revert / fix if needed.

### CODE REVIEW

**Review by pdgendt — COMMENTED** [review]
Nice work! There are some compliance/spelling checks that need fixing.

**Review by ghost — COMMENTED** [review]
Nice! :-) Approved for IEEE 802.15.4 part.

**Comment by jukkar** [comment]
* More CI fixes
* Added more coverage

**Comment by jukkar** [comment]
* Fixed HTTP headers as I missed them earlier

**Review by rlubos — DISMISSED** [review]
Awesome, thanks

**Comment by jukkar** [comment]
* More CI fixes

**Comment by jukkar** [comment]
With these changes we get pretty good coverage for networking API documentation
https://builds.zephyrproject.io/zephyr/pr/72992/api-coverage/zephyr/net/index.html
https://builds.zephyrproject.io/zephyr/pr/72992/api-coverage/zephyr/net/http/index.html

**Review by rlubos — DISMISSED** [review]

File: include/zephyr/net/canbus.h [hunk 1]
```diff
@@ -4,6 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/** @file
+ *  @brief CANBUS definitions.
```
Comment (martinjaeger) [inline include/zephyr/net/canbus.h:8]: Only CAN is an abbreviation (for Controller Area Network), so the correct name is CAN bus. We also have the CAN peripheral driver and this header is only used for the Socket API, so I suggest the following change.

```suggestion
 *  @brief CAN bus socket API definitions.
```

**Review by glarsennordic — DISMISSED** [review]
Looks great, thanks for doing this!

**Comment by jukkar** [comment]
* Updated according to comments

**Review by martinjaeger — APPROVED** [review]
Great, thanks!

**Review by maass-hamburg — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**